src/Utils.hpp

```diff
@@ -0,0 +1,159 @@
+#ifndef GUI_VIZKIT3D_WORLD_SRC_UTILS_HPP_
+#define GUI_VIZKIT3D_WORLD_SRC_UTILS_HPP_
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <QtGui/QImage>
+#include <base/samples/Frame.hpp>
+#include <boost/thread/thread.hpp>
+
+
+/**
+ * Return a std::string with environment variable value
+ *
+ * @param varname: the name of environment variable
+ * @return std::string with environmentn variable value
+ */
+inline std::string getEnv(std::string varname)
+{
+    char *value = getenv(varname.c_str());
+    return (value == NULL) ? "" : std::string(value);
+}
+
+/**
+ * Convert QImage format to frame_mode_t
+ *
+ * @param format: the QImage format
+ * @return base::samples::frame::frame_mode_t
+ */
+inline base::samples::frame::frame_mode_t toFrameMode(QImage::Format format) {
+
+    /**
+     * the QImage formats supported in this function
+     * thes formats have your correspondent frame_mode_t format
+     */
+    static const QImage::Format supported_formats[] = {
+            QImage::Format_Indexed8,
+            QImage::Format_RGB32,
+            QImage::Format_ARGB32,
+            QImage::Format_RGB888,
+    };
+
+    /**
+     * the frame_mode_t supported in this function
+     */
+    static const base::samples::frame::frame_mode_t supported_modes[] = {
+            base::samples::frame::MODE_GRAYSCALE,
+            base::samples::frame::MODE_RGB32,
+            base::samples::frame::MODE_RGB32,
+            base::samples::frame::MODE_BGR,
+            base::samples::frame::MODE_UNDEFINED
+    };
+
+    int totalSupportedFormats = sizeof(supported_modes) / sizeof(base::samples::frame::frame_mode_t);
+
+    int cnt = -1;
+    while ((supported_formats[++cnt] != format) && (cnt < (totalSupportedFormats-1)));
+
+    /**
+     * If the format is not supported, then return value is -1
+     */
+    return supported_modes[cnt];
+}
+
+/**
+ * Copy image pixels from QImage to base::samples::frame::Frame
+ *
+ * @param src: the source image
+ * @param dst: the destionation image
+ * @param flipImage: if is true, flip image in vertical direction
+ */
+inline void cpyQImageToFrame(const QImage& src, base::samples::frame::Frame& dst, bool flipImage = false)
```

Comment:
There are helpers for this kind of operations in perception/frame_helpers. You should have a look there.